internal/payments/service.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package payments
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stripe/stripe-go"
+	"github.com/stripe/stripe-go/client"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+)
+
+// StripeErr is an error for stripe errors
+var StripeErr = errs.Class("stripe error")
+
+// Service is interfaces that defines behavior for working with payments
+type Service interface {
+	CreateCustomer(ctx context.Context, params CreateCustomerParams) (*stripe.Customer, error)
+	GetCustomer(ctx context.Context, customerID string) (*stripe.Customer, error)
+	GetCustomerDefaultPaymentMethod(ctx context.Context, customerID string) (*stripe.PaymentMethod, error)
+	GetCustomerPaymentsMethods(ctx context.Context, customerID string) ([]*stripe.PaymentMethod, error)
+	GetPaymentMethod(ctx context.Context, id string) (*stripe.PaymentMethod, error)
+	CreateProjectInvoice(ctx context.Context, params CreateProjectInvoiceParams) (*stripe.Invoice, error)
+	GetInvoice(ctx context.Context, invoiceID string) (*stripe.Invoice, error)
+}
+
+// StripeService works with stripe network through stripe-go client
+type StripeService struct {
+	log    *zap.Logger
+	client *client.API
+}
+
+// CreateCustomerParams contains info needed to create new stripe customer
+type CreateCustomerParams struct {
+	Email       string
+	Name        string
+	Description string
+	SourceToken string
+}
+
+// CreateProjectInvoiceParams contains info needed to create project invoice
+type CreateProjectInvoiceParams struct {
+	ProjectName     string
+	CustomerID      string
+	PaymentMethodID string
+
+	Storage     float64
+	Egress      float64
+	ObjectCount float64
+
+	StartDate time.Time
+	EndDate   time.Time
+}
+
+// NewService creates new instance of StripeService initialized with API key
+func NewService(log *zap.Logger, apiKey string) *StripeService {
+	stripe.DefaultLeveledLogger = wrapLogger(log)
```

Comment:
stripe.DefaultLeveledLogger = log.Sugar().Named("") should be valid (Didnt test it tho).
It contains the necessary interface functions.